gnome-initial-setup/pages/network/gis-network-page.c

```diff
@@ -625,6 +668,24 @@ gis_network_page_constructed (GObject *object)
     }
   }
 
+  /*
+   * Monitor the network to see if the network gets available without
+   * using this panel.
+   */
+  monitor = g_network_monitor_get_default ();
+
+  g_signal_connect (monitor,
+                    "notify::connectivity",
+                    G_CALLBACK (network_connectivity_changed_cb),
+                    page);
```

Comment:
Should also probably disconnect this signal when the page is destroyed/disposed, since the network monitor is a singleton